steward/src/cork/proposals.rs

```diff
@@ -261,44 +256,51 @@ async fn poll_approved_cork_proposals(state: &mut ProposalThreadState) -> Result
             proposal.proposal_id, encoded_call
         );
 
-        if let Err(err) = schedule_cork(&cellar_id, encoded_call.clone(), block_height).await {
-            match confirm_sheduling(state, &cellar_id, encoded_call, block_height).await {
-                Ok(confirmed) => {
-                    if confirmed {
-                        info!(
-                            "call for cellar {} scheduled for block {} by proposal {}",
-                            cellar_id, block_height, proposal.proposal_id
-                        );
-                        state.increment_proposal_id();
-                        continue;
+        // retry scheduling call on errors up to configured amount of retries
+        let mut attempts = 0;
+        loop {
+            if let Err(err) = schedule_cork(&cellar_id, encoded_call.clone(), block_height).await {
+                error!("error scheduling cork. confirming if the cork was scheduled in case this is a timeout bug: {}", err);
+                match confirm_sheduling(state, &cellar_id, encoded_call.clone(), block_height).await
+                {
+                    Ok(confirmed) => {
+                        if confirmed {
+                            info!(
+                                "call for cellar {} scheduled for block {} by proposal {}",
+                                cellar_id, block_height, proposal.proposal_id
+                            );
+                            state.increment_proposal_id();
+                            break;
+                        }
                     }
-                }
-                Err(err) => {
-                    return Err(proposal_processing_error(format!(
-                        "failed to confirm cork scheduling for proposal {} reason: {}",
-                        proposal.proposal_id, err
-                    )))
-                }
-            };
+                    Err(err) => {
+                        attempts += 1;
+                        if attempts > config.cork.max_scheduling_retries {
+                            return Err(proposal_processing_error(format!(
+                                "failed to schedule cork for proposal {} reason: {}",
+                                proposal.proposal_id, err
```

Comment:
`err` here will be an error returned by `confirm_scheduling` rather than `schedule_cork`, which I think is the error we'd be interested in logging.

Reply:
see line 263, i log the schedule_cork error for this reason and then this one is returned as the final failure message. think it should be changed?

Reply:
I think the only thing is that the log line is indicating the reason for the failure is not seeing the confirmation. This made me realize another issue though: confirm_scheduling returns an `Ok(bool)`. The error match will only happen if the call to confirm the scheduling fails. `Ok(true)` means it was found, but `Ok(false)` means it needs to retry but we aren't handling this condition.

Reply:
good catch